cdap-data-fabric/src/main/java/io/cdap/cdap/data2/dataset2/lib/table/MDSKey.java

```diff
@@ -203,6 +203,11 @@ public Builder add(long part) {
       return this;
     }
 
+    public Builder add(Boolean part) {
+      key = Bytes.add(key, Bytes.toBytes(part));
```

Comment:
the Splitter class also needs to be changed to have a skipBoolean. Also, something should be calling that skipBoolean(), I think in NoSQLStructuredRow there is some logic missing around handling booleans in the row key.